src/WebJobs.Script/DependencyInjection/ScriptStartupTypeLocator.cs

```diff
@@ -323,6 +332,50 @@ void CollectError(Type extensionType, Version minimumVersion, ExtensionStartupTy
             }
         }
 
+        private bool IsValidBindingMatch(ExtensionReference extension, HashSet<string> bindingsSet)
+        {
+            if (extension.Bindings.Count == 0)
+            {
+                return true;
+            }
+
+            // Application Insights uses a special binding value to ensure it's installed based on its App Setting
+            if (extension.Bindings.Any(b => b.Equals("_")))
+            {
+                return IsApplicationInsightsSettingPresent();
+            }
+
+            return extension.Bindings.Intersect(bindingsSet, StringComparer.OrdinalIgnoreCase).Any();
+        }
+
+        private void ValidateApplicationInsightsConfig(List<Type> startupTypes)
+        {
+            if (_isSelfHost)
+            {
+                _logger.LogWarning("In order to use Application Insights in Azure Functions V4 and above, please install the Application Insights Extension.");
```

Comment:
let's make an aka.ms link for this so we can provide more information -- I'll ping you separately.